src/Dfc.CourseDirectory.Core/BinaryStorageProvider/BlobStorageBinaryStorageProvider.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using Azure.Storage.Blobs;
+using Azure;
+
+namespace Dfc.CourseDirectory.Core.BinaryStorageProvider
+{
+    public class BlobStorageBinaryStorageProvider : IBinaryStorageProvider
+    {
+        private readonly BlobContainerClient _blobContainerClient;
+        private bool _containerExists;
+
+        public BlobStorageBinaryStorageProvider(IOptions<BlobStorageBinaryStorageProviderSettings> options)
+        {
+            _blobContainerClient = new BlobContainerClient(options.Value.ConnectionString, options.Value.ContainerName);
+        }
+
+        public async Task<bool> TryDownloadFile(string path, Stream destination)
+        {
+            await EnsureContainerExists();
+
+            var blob = _blobContainerClient.GetBlobClient(path);
+
+            try
+            {
+                await blob.DownloadToAsync(destination);
+                return true;
+            }
+            catch (RequestFailedException ex) when (ex.Status == 404)
```

Comment:
I prefer to use `StatusCodes.Status404NotFound` for readability.

Reply:
That's an ASP.NET Core type - don't want to introduce a dependency on that in this project

Reply:
You're quite right, I though it was standard! Could use `System.Net.HttpStatusCode.NotFound` then?